src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter_time.cs

```diff
@@ -1,69 +1,39 @@
-
-        static readonly byte[][] s_dayNamesUtf8 = {
-            new Utf8String("Sun, ").Bytes.ToArray(),
-            new Utf8String("Mon, ").Bytes.ToArray(),
-            new Utf8String("Tue, ").Bytes.ToArray(),
-            new Utf8String("Wed, ").Bytes.ToArray(),
-            new Utf8String("Thu, ").Bytes.ToArray(),
-            new Utf8String("Fri, ").Bytes.ToArray(),
-            new Utf8String("Sat, ").Bytes.ToArray(),
-        };
-        static readonly byte[][] s_monthNamesUtf8 = {
-            new Utf8String("Jan").Bytes.ToArray(),
-            new Utf8String("Feb").Bytes.ToArray(),
-            new Utf8String("Mar").Bytes.ToArray(),
-            new Utf8String("Apr").Bytes.ToArray(),
-            new Utf8String("May").Bytes.ToArray(),
-            new Utf8String("Jun").Bytes.ToArray(),
-            new Utf8String("Jul").Bytes.ToArray(),
-            new Utf8String("Aug").Bytes.ToArray(),
-            new Utf8String("Sep").Bytes.ToArray(),
-            new Utf8String("Oct").Bytes.ToArray(),
-            new Utf8String("Nov").Bytes.ToArray(),
-            new Utf8String("Dec").Bytes.ToArray(),
-        };
-        static readonly byte[] s_gmtUtf8Bytes = new Utf8String(" GMT").Bytes.ToArray();
-
-        static readonly TextFormat D2 = new TextFormat('D', 2);
-        static readonly TextFormat D4 = new TextFormat('D', 4);
-        static readonly TextFormat D7 = new TextFormat('D', 7);
-        static readonly TextFormat G = new TextFormat('G'); 
-        static readonly TextFormat t = new TextFormat('t'); 
-        const int FractionalTimeScale = 10000000;
+        internal static readonly TimeSpan NullOffset = TimeSpan.MinValue;
 
         public static bool TryFormat(this DateTimeOffset value, Span<byte> buffer, out int bytesWritten, TextFormat format = default(TextFormat), TextEncoder encoder = null)
         {
+            TimeSpan offset = NullOffset;
             if (format.IsDefault)
             {
                 format.Symbol = 'G';
+                offset = value.Offset;
             }
+
             Precondition.Require(format.Symbol == 'R' || format.Symbol == 'O' || format.Symbol == 'G');
 
-            encoder = encoder== null ? TextEncoder.Utf8 : encoder;
+            encoder = encoder == null ? TextEncoder.Utf8 : encoder;
 
             switch (format.Symbol)
             {
                 case 'R':
                     return TryFormatDateTimeRfc1123(value.UtcDateTime, buffer, out bytesWritten, encoder);
+
                 case 'O':
-                    return TryFormatDateTimeFormatO(value.UtcDateTime, false, buffer, out bytesWritten, encoder);
+                    return TryFormatDateTimeFormatO(value.DateTime, value.Offset, buffer, out bytesWritten, encoder);
+
                 case 'G':
-                    return TryFormatDateTimeFormatG(value.DateTime, buffer, out bytesWritten, encoder);
+                    return TryFormatDateTimeFormatG(value.DateTime, offset, buffer, out bytesWritten, encoder);
```

Comment:
I have tests for the "default" case and this is passing based on what DateTimeOffset.ToString returns. For some reason, the system DateTime.ToString "default" (though it is documented as 'G') actually does something slightly different than explicitly using 'G'. I don't claim to understand why, but these APIs now follow exactly what the system does in all the cases we support.